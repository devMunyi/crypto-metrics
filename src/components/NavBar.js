import React, { useState, useEffect } from 'react';
import { useLocation, NavLink } from 'react-router-dom';
import { FaChevronLeft, FaMicrophone } from 'react-icons/fa';
import { BsFillGearFill } from 'react-icons/bs';

const NavBar = () => {
  const location = useLocation()?.pathname;

  const [currentPath, setCurrentPath] = useState('');

  useEffect(() => {
    setCurrentPath(location);
  }, [location]);

  return (
    <nav className="mb-2 d-flex justify-content-between align-items-center container-fluid headline">
      <div>
        {currentPath === '/' ? (
          '2022'
        ) : (
          <NavLink style={{ color: '#fff' }} to="/">
            <FaChevronLeft title="Click to go back" />
          </NavLink>
        )}
      </div>
      <div>{currentPath === '/' ? 'Home' : 'Details'}</div>
      <div
        style={{ width: '60px' }}
        className="d-flex justify-content-between align-items-center"
      >
        <div className="mic">
          <FaMicrophone />
        </div>
        <div className="settings">
          <BsFillGearFill />
        </div>
      </div>
    </nav>
  );
};

export default NavBar;
