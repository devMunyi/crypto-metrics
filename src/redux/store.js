import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import cryptoReducer from './reducers';

export default configureStore(
  {
    reducer: {
      crypto: cryptoReducer,
    },
  },
  applyMiddleware(logger),
);
