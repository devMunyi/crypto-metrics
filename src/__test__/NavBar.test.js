import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import NavBar from '../components/NavBar';

/* eslint-disable comma-dangle */
it('navbar renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <NavBar />
        </Router>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
