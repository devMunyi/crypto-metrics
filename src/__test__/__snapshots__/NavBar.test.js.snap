// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar renders correctly 1`] = `
<nav
  className="mb-2 d-flex justify-content-between align-items-center container-fluid headline"
>
  <div>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "color": "#fff",
        }
      }
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 320 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Click to go back
        </title>
        <path
          d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
        />
      </svg>
    </a>
  </div>
  <div>
    Details
  </div>
  <div
    className="d-flex justify-content-between align-items-center"
    style={
      Object {
        "width": "60px",
      }
    }
  >
    <div
      className="mic"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 352 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
        />
      </svg>
    </div>
    <div
      className="settings"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.405 1.05c-.413-1.4-2.397-1.4-2.81 0l-.1.34a1.464 1.464 0 0 1-2.105.872l-.31-.17c-1.283-.698-2.686.705-1.987 1.987l.169.311c.446.82.023 1.841-.872 2.105l-.34.1c-1.4.413-1.4 2.397 0 2.81l.34.1a1.464 1.464 0 0 1 .872 2.105l-.17.31c-.698 1.283.705 2.686 1.987 1.987l.311-.169a1.464 1.464 0 0 1 2.105.872l.1.34c.413 1.4 2.397 1.4 2.81 0l.1-.34a1.464 1.464 0 0 1 2.105-.872l.31.17c1.283.698 2.686-.705 1.987-1.987l-.169-.311a1.464 1.464 0 0 1 .872-2.105l.34-.1c1.4-.413 1.4-2.397 0-2.81l-.34-.1a1.464 1.464 0 0 1-.872-2.105l.17-.31c.698-1.283-.705-2.686-1.987-1.987l-.311.169a1.464 1.464 0 0 1-2.105-.872l-.1-.34zM8 10.93a2.929 2.929 0 1 1 0-5.86 2.929 2.929 0 0 1 0 5.858z"
        />
      </svg>
    </div>
  </div>
</nav>
`;
